core/src/test/java/org/apache/iceberg/io/inmemory/InMemoryFileIOTest.java

```diff
@@ -0,0 +1,256 @@
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class InMemoryFileIOTest {
+
+  private InMemoryFileIO fileIO;
+  private InMemoryFileStore store;
+
+  @BeforeEach
+  public void before() {
+    fileIO = new InMemoryFileIO();
+    store = fileIO.getStore();
+  }
+
+  @Test
+  public void testGetStore() {
+    Assertions.assertThat(store).isNotNull();
+  }
+
+  @Test
+  public void testDeleteFile() {
+    Assertions.assertThat(store.exists("file1")).isFalse();
+
+    // Create the file
+    store.put("file1", ByteBuffer.wrap(new byte[0]));
+    Assertions.assertThat(store.exists("file1")).isTrue();
+
+    // Delete the file
+    fileIO.deleteFile("file1");
+    // Verify that the file has been deleted
+    Assertions.assertThat(store.exists("file1")).isFalse();
+
+    Assertions.assertThatThrownBy(() -> fileIO.deleteFile("nonExisting"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("File at location 'nonExisting' does not exist");
+  }
+
+  @Test
+  public void testNewInputFile() throws IOException {
+    String fileName = "file1";
+
+    store.put(fileName, ByteBuffer.wrap("data1".getBytes(UTF_8)));
+
+    InputFile inputFile = fileIO.newInputFile(fileName);
+    Assertions.assertThat(inputFile).isNotNull();
+
+    Assertions.assertThat(inputFile.exists()).isTrue();
+    Assertions.assertThat(inputFile.location()).isEqualTo(fileName);
+    Assertions.assertThat(inputFile.getLength()).isEqualTo(5);
+
+    try (SeekableInputStream inputStream = inputFile.newStream()) {
+      Assertions.assertThat(inputStream).isNotNull();
+
+      // Test read()
+      Assertions.assertThat(inputStream.getPos()).isEqualTo(0);
+      Assertions.assertThat(inputStream.read()).isEqualTo('d');
+
+      // Test read(byte[], index, len)
+      byte[] dataRead = new byte[3];
+      Assertions.assertThat(inputStream.read(dataRead, 0, 2)).isEqualTo(2);
+      Assertions.assertThat(new String(dataRead, 0, 2, UTF_8)).isEqualTo("at");
+    }
+  }
+
+  @Test
+  public void testSeek() throws IOException {
+    String fileName = "file1";
+
+    // Number of rows
+    int numRowsInChunk = 10;
+
+    // Emulate parquet file structure
+    ByteBuffer buffer = ByteBuffer.allocate(1024);
+    // Add magic number
+    buffer.put("par1".getBytes(UTF_8));
+    // Add column chunk 1 for int32
+    IntStream.range(0, numRowsInChunk).forEach(buffer::putInt);
+    // Add column chunk 2 for float64
+    IntStream.range(0, numRowsInChunk).forEach(buffer::putDouble);
+    // Add metadata footer (index to the start of chunks)
+    buffer.putInt(4);
+    buffer.putInt(4 + numRowsInChunk * 4);
+    // Add footer size
+    buffer.putInt(8);
+    // Add magic number
+    buffer.put("par1".getBytes(UTF_8));
+
+    // Put the data in the store
+    buffer.flip();
+    store.put(fileName, buffer);
+
+    // magic number, chunk1, chunk2, footer, footer size, magic number
+    int expectedFileSize = 4 + numRowsInChunk * 4 + numRowsInChunk * 8 + 8 + 4 + 4;
```

Comment:
Nit: Some of these might be better off as constants. But since it's all in the same method and we'll defined above I'm ok either way.